### ISSUE TITLE
btrfs builds fail due to already existing path

### ISSUE THREAD

**Issue by dud225** [systemd/mkosi#396]
Hello

```bash
mkosi -t subvolume -o debian-test -r testing -d debian
‣ Processing default...
[...]
‣ Detaching namespace...
‣ Detaching namespace complete.
‣ Setting up package cache...
‣ Setting up package cache /var/lib/machines/.mkosi-hvtzmez6 complete.
‣ Setting up temporary workspace.
‣ Temporary workspace in /var/lib/machines/.mkosi-t0yppp_i is now set up.
‣ Running second (final) stage...
‣ Mounting image...
‣ Mounting image complete.
‣ Setting up basic OS tree...
ERROR: target path already exists: /var/lib/machines/.mkosi-t0yppp_i/root
‣ Unmounting image...
‣ Unmounting image complete.
Traceback (most recent call last):
  File "mkosi", line 4872, in <module>
    main()
  File "mkosi", line 4868, in main
    run_verb(a)
  File "mkosi", line 4843, in run_verb
    build_stuff(args)
  File "mkosi", line 4667, in build_stuff
    raw, tar, root_hash = build_image(args, workspace, do_run_build_script=False, cleanup=True)
  File "mkosi", line 4471, in build_image
    prepare_tree(args, workspace.name, do_run_build_script, cached)
  File "/usr/lib/python3.7/contextlib.py", line 74, in inner
    return func(*args, **kwds)
  File "mkosi", line 1083, in prepare_tree
    btrfs_subvol_create(os.path.join(workspace, "root"))
  File "mkosi", line 467, in btrfs_subvol_create
    run(["btrfs", "subvol", "create", path], check=True)
  File "mkosi", line 76, in run
    return subprocess.run(cmdline, **kwargs)
  File "/usr/lib/python3.7/subprocess.py", line 512, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['btrfs', 'subvol', 'create', '/var/lib/machines/.mkosi-t0yppp_i/root']' returned non-zero exit status 1.
```

I've tracked down this issue to this [commit](https://github.com/systemd/mkosi/commit/0a9928d5506fa94d866da4be64dc1bcf5f9d97d0#diff-6980c629516e4165bfaf96be8e5003cf).
@lucasdemarchi I'm not sure about the proper fix to bring. I've checked that adding back the condition fixes the issue, however if I understand properly your commit the idea is to mount something on `root` so that the later `umount` cleans out all things that may have been mounted on top this tree. Any suggestion about the proper way to fix this ?

**Comment by stejoo** [comment]
I can confirm the bug: I also ran into this a couple of months ago. Attempted to debug at the time but wasn't able to deduce a fix at that moment. End of year rush came and I forgot all about it, have been using tar method since.

tl;dr: +1 can confirm this bug.
